ts/WoltLabSuite/Core/Acp/Ui/User/Action/Handler/Ban.ts

```diff
@@ -0,0 +1,144 @@
+import { DialogCallbackSetup } from "../../../../../Ui/Dialog/Data";
+import * as Language from "../../../../../Language";
+import * as Ajax from "../../../../../Ajax";
+import UiDialog from "../../../../../Ui/Dialog";
+
+/**
+ * @author  Joshua Ruesweg
+ * @copyright  2001-2021 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Acp/Ui/User/Action/Handler
+ * @since       5.5
+ */
+export class BanHandler {
+  private userIDs: number[];
+  private banCallback: () => void;
+
+  public constructor(userIDs: number[]) {
+    this.userIDs = userIDs;
+  }
+
+  ban(callback: () => void): void {
+    // Save the callback for later usage.
+    // We cannot easily give the callback to the dialog.
+    this.banCallback = callback;
+
+    UiDialog.open(this);
+  }
+
+  unban(callback: () => void): void {
+    Ajax.api({
+      _ajaxSetup: () => {
+        return {
+          data: {
+            actionName: "unban",
+            className: "wcf\\data\\user\\UserAction",
+            objectIDs: this.userIDs,
+          },
+        };
+      },
+      _ajaxSuccess: callback,
+    });
+  }
+
+  private banSubmit(reason: string, userBanExpires: string): void {
+    Ajax.api({
+      _ajaxSetup: () => {
+        return {
+          data: {
+            actionName: "ban",
+            className: "wcf\\data\\user\\UserAction",
+            objectIDs: this.userIDs,
+            parameters: {
+              banReason: reason,
+              banExpires: userBanExpires,
+            },
+          },
+        };
+      },
+      _ajaxSuccess: this.banCallback,
+    });
+  }
+
+  _dialogSetup(): ReturnType<DialogCallbackSetup> {
+    return {
+      id: "userBanHandler",
```

Comment:
As each `BanAction` has its own ban handler, could this general id cause issues if multiple `BanAction` instances exist? (same applies to using IDs below)